app/models/webhook.rb

```diff
@@ -1,11 +1,19 @@
 class Webhook < ActiveRecord::Base
   has_soft_deletion default_scope: true
-  validates :branch, uniqueness: { scope: [ :stage], conditions: -> { where("deleted_at IS NULL") }, message: "one webhook per (stage, branch) combination." }
+  validates :branch, uniqueness: { scope: [ :stage ], conditions: -> { where("deleted_at IS NULL") }, message: "one webhook per (stage, branch) combination." }
 
   belongs_to :project
   belongs_to :stage
 
+  SOURCES = Rails.root.join('app','controllers', 'integrations').children(false).map do |controller_path|
```

Comment:
that sounds nice, loading more controllers is fine in dev/test as long as we don't load them at startup (triggers load of all kinds of other things)

the code can be test/dev only since production will eager load ... maybe be in an `if Rails.application.eager_load` block